tests/test_math_interpolator.hh

```diff
@@ -0,0 +1,158 @@
+/**
+ * file   test_math_interpolator.hh
+ *
+ * @author  Alexander Goscinski <alexander.goscinski@epfl.ch>
+ *
+ * @date   26 August 2019
+ *
+ * @brief Tests the implementation of interpolator
+ *
+ * Copyright © 2019  Alexander Goscinski, COSMO (EPFL), LAMMM (EPFL)
+ *
+ * rascal is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3, or (at
+ * your option) any later version.
+ *
+ * rascal is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GNU Emacs; see the file COPYING. If not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+#include "tests.hh"
+#include "test_math.hh"
+#include "math/interpolator.hh"
+#include "representations/calculator_spherical_expansion.hh"
+
+#ifndef TESTS_TEST_MATH_INTERPOLATOR_HH_
+#define TESTS_TEST_MATH_INTERPOLATOR_HH_
+
+namespace rascal {
+  using Vector_t = math::Vector_t;
+  using Matrix_t = math::Matrix_t;
+  using Vector_Ref = math::Vector_Ref;
+  using Matrix_Ref = math::Matrix_Ref;
+
+  using AbsoluteErrorMethod = math::ErrorMethod<math::ErrorMetric_t::Absolute>;
+  using RelativeErrorMethod = math::ErrorMethod<math::ErrorMetric_t::Relative>;
+
+  static double convert_to_scalar(const double x) { return x; }
+  static double convert_to_scalar(const math::Matrix_t & mat) {
+    return mat(0, 0);
+  }
+
+  // These functions calculate the error between the interpolated function and
+  // the interpolation on a set of reference points for different
+
+  template <class ErrorMethod, class ScalarInterpolator>
+  static double compute_intp_error(std::shared_ptr<ScalarInterpolator> & intp,
+                                   std::function<double(double)> ref_func,
```

Comment:
I am not 100% familiar with this notation of the function, just to understand: the template parameter here means the function takes a `double` as an argument and also returns a `double`, right?

And can these be `const`?

Reply:
>  just to understand: the template parameter here means the function takes a double as an argument and also returns a double, right?

Yes. And by const do you mean `std::function<const double(const double)>`? If so, they are not needed.

Reply:
I mean _before_ the type as in
`(const std::shared_ptr<ScalarInterpolator> & intp,...`

Reply:
ok, sorry I got confused with the above discussion on using const on parameter by value 😃